### ISSUE TITLE
MESMER-X: thoughts on `fg_fun_scale` 

### ISSUE THREAD

**Issue by veni-vidi-vici-dormivi** [MESMER-group/mesmer#583]
TLDR: the current function to optimize for the scale is not optimal for all distributions but it is not worth changing it for now.

For a preliminary first guess for the scale of a distribution,  we optimize the scale parameter such that it is close to the absolute deviations of the samples from the location: 

https://github.com/MESMER-group/mesmer/blob/72d83a5b9567da90f7de12a2772e1b16a7fefd3d/mesmer/mesmer_x/train_l_distrib_mesmerx.py#L1164-L1171

I wondered if instead of using absolute deviations, we should go for the squared deviations. It turns out that either option works better for specific distributions: 
 - using squared deviations works better for distributions for whom the scale parameter is closely related to the squared deviations from the location (duh). This is for example the normal distribution where the scale parameter is equal to the variance of the samples (when the number of samples approaches inf). Others include 
 - using absolute deviations works better for distributions for whom the scale parameter is more closely related to the absolute deviations. This includes the GEV and the Laplace distribution.

I tested it and indeed the first guess after this step improves for a normal distribution when switching to the squared deviations but worsens for the GEV. However, we do another fit for all parameters after this where the first guess is further improved and this worked well for my examples. Thus, **I would not change anything for now**, I just wanted to write this down to show the tradeoffs we are facing.
